src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -208,6 +220,52 @@ Transactor::calculateBaseFee(ReadView const& view, STTx const& tx)
     return baseFee + (signerCount * baseFee);
 }
 
+TER
+Transactor::checkPermissions(
+    ReadView const& view,
+    STTx const& tx,
+    std::unordered_set<GranularPermissionType>& permissions)
+{
+    if (!tx.isFieldPresent(sfOnBehalfOf) ||
+        !view.exists(keylet::account(tx[sfOnBehalfOf])))
+        return terNO_ACCOUNT;
+
+    auto const accountPermissionKey =
+        keylet::accountPermission(tx[sfOnBehalfOf], tx[sfAccount]);
+    auto const sle = view.read(accountPermissionKey);
+    if (!sle)
+        return tecNO_PERMISSION;
+
+    auto const permissionArray = sle->getFieldArray(sfPermissions);
+    auto const transactionType = tx.getTxnType();
+
+    for (auto const& permission : permissionArray)
+    {
+        auto const permissionValue = permission[sfPermissionValue];
+        if (permissionValue == transactionType + 1)
```

Comment:
probably shouldn't hardcode the logic `transactionType + 1` here. we should prob have a function named `TransactionPermissionValue(TxType)` that does this and use it across all the places (I believe STParsedJSON is  hardcoded too)